### PR TITLE
internal: clean up all usages of grpclog, and replace with component

### PR DESCRIPTION
**Pull request by menghanl**

### CODE REVIEW

File: internal/binarylog/binarylog_end2end_test.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+var grpclogLogger = grpclog.Component("binarylog")
```
Comment (easwars) [inline internal/binarylog/binarylog_end2end_test.go:43]: Do we want to do this in a test file?

File: interop/xds/server/server.go [hunk 2]
```diff
@@ -36,6 +36,8 @@ var (
 	port     = flag.Int("port", 8080, "The server port")
 	serverID = flag.String("server_id", "go_server", "Server ID included in response")
 	hostname = getHostname()
+
+	logger = grpclog.Component("interop")
```
Comment (easwars) [inline interop/xds/server/server.go:40]: Would it make sense to have a `logging.go` file in `intertop/` and define the logger there?

File: internal/binarylog/binarylog_end2end_test.go (hunk 1 above)
Comment (menghanl) [reply internal/binarylog/binarylog_end2end_test.go:43]: This test is in a separate package, so it either defines it's own logger for the component, or the logger needs to be exported.

File: interop/xds/server/server.go (hunk 2 above)
Comment (menghanl) [reply interop/xds/server/server.go:41]: Creating this component only takes a reference to the existing component (if it always exists), so the overhead is small.

The `interop` package has too many things that aren't needed by all tests.

**Comment by easwars** [comment]
Looks good to me except for the "sts" component.

**Pull request merged**